### DISCUSSION TITLE
Adding 29bit canid support?

### DISCUSSION THREAD

**Discussion by bayti** [jglim/CaesarSuite#46] [General]
Possible adding 29bit CANID Support !

**Comment by Feezex** [comment]
is there any purpose?

**Reply by bayti** [reply]
yes, unable to connect to truck control units. only 11 bits are supported. trucks and buses work with 29 bit can id.

**Reply by jglim** [reply]
Unfortunately this isn't supported at the hardware level. 29-bit devices typically use J1939 which isn't supported by the J2534 communication standard. [This earlier issue](https://github.com/jglim/CaesarSuite/issues/35) discusses about 29-bit support.

**Reply by bayti** [reply]
Hello @jglim jglim, i am using already 29 bit devices with j2534 communication interface. Example. Chipsoft(j2534-24V)--> pcmflash.ru (software(flash) read and write Scania Ecu's.) i think , its communicate ISO15765-2 Extended ID. Please inform me, if i misunderstand anything!

[Peak UDS documentation ](https://www.peak-system.com/produktcd/Pdf/English/PCAN-UDS-API_UserMan_eng.pdf)

------------------------
<img width="698" alt="Ekran Resmi 2022-02-21 23 58 35" src="https://user-images.githubusercontent.com/39216096/155025697-d7638393-1f48-4d88-886b-c8ad58d45ccd.png">
-----------------------
<img width="695" alt="Ekran Resmi 2022-02-22 00 01 32" src="https://user-images.githubusercontent.com/39216096/155025878-9fde60f8-2a18-4edc-b6ae-7a1a6255ac62.png">

**Reply by jglim** [reply]
If those ECUs uses regular ISO15765-2, then the connection should (in theory) be configured correctly, as I specified `ConnectFlag.CAN_ID_BOTH` during the device's initialization:

https://github.com/jglim/CaesarSuite/blob/c9ec0d258d370f8c6ee110dc0fd3267bdc616bc8/Caesar/Diogenes/ECUConnection.cs#L275

However, if it does not work out of the box, it is going to be difficult and time-consuming for me to diagnose the issue as I do not have a compatible 29-bit bus/truck ECU. As a minimum, working/broken CAN traces of would be helpful for diffing/comparison.
